src/java/org/apache/cassandra/io/sstable/CQLSSTableWriter.java

```diff
@@ -648,6 +696,29 @@ private Types createTypes(String keyspace)
             return builder.build();
         }
 
+        private Indexes createIndexes(TableMetadata tableMetadata)
+        {
+            ClientState state = ClientState.forInternalCalls();
+            Indexes.Builder builder = Indexes.builder();
+
+            for (CreateIndexStatement.Raw createIndexStatement : indexStatements)
+            {
+                CreateIndexStatement statement = createIndexStatement.prepare(state);
+                statement.validate(ClientState.forInternalCalls());
+
+                List<IndexTarget> indexTargets = statement.rawIndexTargets.stream()
+                                                                          .map(raw -> raw.prepare(tableMetadata))
+                                                                          .collect(Collectors.toList());
+
+                builder.add(IndexMetadata.fromIndexTargets(indexTargets,
+                                                           statement.indexName,
+                                                           IndexMetadata.Kind.CUSTOM,
+                                                           statement.attrs.getOptions()));
+            }
+
+            return builder.build();
+        }
```

Comment:
If we in stead do something like `applyIndexes` we can do this in a way I think is a bit cleaner, and will pick up any changes to the index statement logic, by using `CreateIndexStatement#apply`

```suggestion
                /**
         * Applies any provided index definitions to the target table
         * @param ksm the KeyspaceMetadata object that has the table defined
         * @return an updated TableMetadata instance with the indexe create statements applied
         */
        private TableMetadata applyIndexes(KeyspaceMetadata ksm)
        {
            ClientState state = ClientState.forInternalCalls();
            Keyspaces keyspaces = Keyspaces.of(ksm);
            for (CreateIndexStatement.Raw statement: indexStatements)
            {
                keyspaces = statement.prepare(state).apply(keyspaces);
            }
            return keyspaces.get(ksm.name).get()
                   .tables.get(schemaStatement.table()).get();
        }
```

And then, in `#build` we end up with something like this:


``` java
                if (tableMetadata == null)
                {
                    Types types = createTypes(keyspaceName);
                    Schema.instance.transform(SchemaTransformations.addTypes(types, true));
                    tableMetadata = createTable(types);
                    if (!indexStatements.isEmpty())
                        tableMetadata = applyIndexes(ksm.withSwapped(ksm.tables.with(tableMetadata)));
                    Schema.instance.transform(SchemaTransformations.addTable(tableMetadata, true));
                }
```